apps/aechannel/src/aesc_fsm_min_depth_watcher.erl

```diff
@@ -89,87 +100,179 @@ terminate(_Reason, _St) ->
-    TopHeight = top_height(),
-    St1 =
-        case aec_chain:get_channel(ChId) of
-            {ok, Ch} ->
-                case aesc_channels:is_active(Ch) of
-                    true ->
-                        St;
-                    false ->
-                        St#{closes_at => aesc_channels:closes_at(Ch)}
-                end;
-            {error, _} ->
-                %% Set closing time to current top height, wait for min_depth
-                St#{closes_at => TopHeight}
-        end,
-    {noreply, St1};
-check_status(#{tx_hash := undefined} = St) ->
-    watch_for_chain_transaction(St);
-check_status(#{parent := Parent, chan_id := ChanId,
-               tx_hash := TxHash, type := Type, min_depth := MinDepth} = St) ->
-    lager:debug("check_status(~p)", [Parent]),
-    case min_depth_achieved(TxHash, MinDepth) of
+check_status(#st{requests = Reqs} = St) ->
+    check_status(Reqs, St, #{}, []).
+
+check_status([Req|Reqs], St, Cache, Acc) ->
+    case check_status_(Req, St, Cache) of
+        {done, #{} = Cache1} ->
+            check_status(Reqs, St, Cache1, Acc);
+        {done, #st{} = St1, #{} = Cache1} ->
+            check_status(Reqs, St1, Cache1, Acc);
+        {#{} = Req1, Cache1} ->
+            check_status(Reqs, St, Cache1, [Req1|Acc])
+    end;
+check_status([], St, _, Acc) ->
+    St#st{requests = lists:reverse(Acc)}.
+
+check_status_(#{mode := close, closes_at := H, min_depth := Min} = R,
+              #st{chan_id = ChId}, Cache) ->
+    {HasDepth, Cache1} = determine_depth_(H, Min, Cache),
+    case HasDepth of
         true ->
+            #{callback_mod := Mod, parent := Parent} = R,
+            Mod:minimum_depth_achieved(Parent, ChId, close, undefined),
+            {done, Cache1};
+        _ ->
+            {R, Cache1}
+    end;
+check_status_(#{mode := close, parent := Parent} = R, #st{chan_id = ChId}, C) ->
+    lager:debug("check_status(type = close, parent = ~p)", [Parent]),
+    {TopHeight, C1} = top_height(C),
+    {Status, C2} = channel_status(ChId, C1),
+    case Status of
+        #{ is_active := true } ->
+            {R, C2};
+        #{ closes_at := ClosesAt } ->
+            {R#{ closes_at => ClosesAt }, C2};
+        undefined ->
+            %% Set closing time to current top height, wait for min_depth
+            {R#{ closes_at => TopHeight }, C2}
+    end;
+check_status_(#{mode := watch} = R, St, C) ->
+    watch_for_channel_closing(R, St, C);
+check_status_(#{mode := tx_hash, tx_hash := TxHash, parent := Parent,
+                type := Type, min_depth := MinDepth} = R,
+              #st{chan_id = ChanId}, C) ->
+    lager:debug("check_status(tx_hash = ~p, parent = ~p)", [TxHash, Parent]),
+    case min_depth_achieved(TxHash, MinDepth, C) of
+        {true, C1} ->
             lager:debug("min_depth achieved", []),
-            case aesc_fsm:minimum_depth_achieved(
+            #{ callback_mod := Mod } = R,
+            case Mod:minimum_depth_achieved(
```

Comment:
A remnant from an idea that I think is now obsolete, and the fact that dialyzer is unhappy if a case clause exists but isn't reached, because no callback makes use of it.